integration/src/test/scala/zio/jdbc/PgSpec.scala

```diff
@@ -1,33 +1,43 @@
 package zio.jdbc
 
-import org.testcontainers.containers.PostgreSQLContainer
-import org.testcontainers.containers.PostgreSQLContainerProvider
-import zio.ZLayer
-import zio.ZIO
+import org.testcontainers.containers.{ PostgreSQLContainer, PostgreSQLContainerProvider }
 import zio.test.ZIOSpec
+import zio.{ ZIO, ZLayer }
+
+import java.util.TimeZone
 
 abstract class PgSpec extends ZIOSpec[ZConnectionPool] {
-  def bootstrap: ZLayer[Any, Throwable, ZConnectionPool] =
-    (ZLayer.scoped {
-      ZIO.fromAutoCloseable {
-        ZIO.attemptBlocking {
-          val provider = new PostgreSQLContainerProvider
-          val db       = provider.newInstance()
-          db.start()
-          db.asInstanceOf[PostgreSQLContainer[Nothing]]
+
+  /**
+   * We need to set the JVM default timezone to UTC before doing anything else
+   * because this default time zone will be picked up by JDBC (See [[org.postgresql.jdbc.TimestampUtils.getDefaultTz]]
+   * and will be used to convert timestamps to UTC before to send them to the DB.
+   * When the computer on which tests are running is using UTC (like in CI, for example), that'll not change anything.
+   * When the computer on which tests are running is using another timezone, the `java.time.*` tests will fail because the saved timestamps will be converted to UTC from the computer timezone.
+   * So, we need to set the VM default timezone to UTC so that the tests are executed in the exact same way and produce the exact same results in the CI or on your local computer.
+   */
+  override def bootstrap: ZLayer[Any, Throwable, ZConnectionPool] =
+    ZLayer.succeed(TimeZone.setDefault(TimeZone.getTimeZone("UTC"))) >>>
```

Comment:
I tried to document as much as possible just above why we need this as it's not obvious (and took me 4h to find)

Reply:
@guizmaii I think this can't be right. Why? Because tests should ensure invariants. In the invariant for date/time related tests should be, that when I write a value to the db and read it, it should be the same value **independent** of the system time.
Else that would mean, that I save a date on my server with for example Berlin timezone and then read it and show it in the frontend and suddenly have a timestamp that is not Berlin timezone?
That seems odd. Like something I'd expect my library to handle. Not me.

Reply:
Why woud your server be in the Berlin timezone?

Reply:
See https://github.com/zio/zio-jdbc/pull/182